cmd/agent/main.go

```diff
@@ -50,9 +52,19 @@ func main() {
 
 			builder := &app.Builder{}
 			builder.InitFromViper(v)
+			tchanRep := tchannel.NewBuilder().InitFromViper(v)
+			mFactory, err := builder.GetMetricsFactory()
+			if err != nil {
+				logger.Fatal("Could not create metrics", zap.Error(err))
+			}
+			r, err := tchanRep.CreateReporter(mFactory, logger)
```

Comment:
I wonder if it would make more sense to encapsulate "tchannel collector proxy" that will either implement or at least expose both Reporter and ClientConfigManager APIs. It would make the code in main() simpler (and code itself can be unit tested), and will be easier to switch for gRPC implementation.

Right now `httpserver.NewCollectorProxy()` below is leaking tchannel detail outside of tchannel package.